CA1816 in `NamedPipeClient.Dispose()`

In `PowerShell-CI-linux`: https://dev.azure.com/powershell/PowerShell/_build/results?buildId=111689&view=logs&j=8579d1dd-da9d-5760-afc4-fb4c05cf5d6d&t=236f801b-8fac-50c5-f817-425a182daa54&l=6000

```log
/home/vsts/work/1/s/test/tools/NamedPipeConnection/src/code/NamedPipeConnection.cs(61,13): warning CA1816: NamedPipeClient.Dispose() calls GC.SuppressFinalize(object), a method that is typically only called within an implementation of 'IDisposable.Dispose'. Refer to the IDisposable pattern for more information. [/home/vsts/work/1/s/test/tools/NamedPipeConnection/src/code/Microsoft.PowerShell.NamedPipeConnection.csproj::TargetFramework=net7.0]
```

cc: @PaulHigin

--- a/test/tools/NamedPipeConnection/src/code/NamedPipeConnection.cs
+++ b/test/tools/NamedPipeConnection/src/code/NamedPipeConnection.cs
@@ -22,7 +22,7 @@ namespace Microsoft.PowerShell.CustomNamedPipeConnection
     /// a client side named pipe object that can connect to a running PowerShell 
     /// process by its process Id.
     /// </summary>
-    internal sealed class NamedPipeClient
+    internal sealed class NamedPipeClient : IDisposable
     {
         #region Members
 

